lib/camera.ts

```diff
@@ -190,53 +192,62 @@ class Camera {
     })
   }
 
-  loadDriver() {
-      try {
-          utils.execSync("sudo modprobe bcm2835-v4l2"); // only on PI, and not needed with USB Camera
-      } catch (err) {}
-  }
-  
-  unloadDriver(){
-      try {
-          utils.execSync("sudo modprobe -r bcm2835-v4l2");
-      } catch (err) {}
-  }
-
   setupCamera() {
     v4l2ctl.SetPixelFormat(v4l2ctl.Pixelformat.H264)
-    v4l2ctl.SetResolution(this.settings.resolution);
-    v4l2ctl.SetFrameRate(this.settings.framerate);
     v4l2ctl.SetPriority(v4l2ctl.ProcessPriority.record);
     v4l2ctl.ReadFromFile();
-    v4l2ctl.ApplyControls();
+    this.setSettings(this.settings);
   }
 
-  setSettings(newsettings: CameraSettingsParameter) {
-    v4l2ctl.SetResolution(newsettings.resolution);
-    v4l2ctl.SetFrameRate(newsettings.framerate);
+  setSettings(newsettings: CameraSettings) {
+    const requireRestartParams = [c => c.resolution.Height, c => c.resolution.Width, c => c.framerate];
+    const requiresRestart = requireRestartParams.some(f => f(this.settings) !== f(newsettings));
+    Object.assign(this.settings, newsettings);
 
     v4l2ctl.Controls.CodecControls.video_bitrate.value = newsettings.bitrate * 1000;
     v4l2ctl.Controls.CodecControls.video_bitrate_mode.value = newsettings.quality > 0 ? 0 : 1;
-    v4l2ctl.Controls.CodecControls.h264_i_frame_period.value = this.settings.forceGop ? v4l2ctl.Controls.CodecControls.h264_i_frame_period.value : newsettings.gop;
+    v4l2ctl.Controls.CodecControls.h264_i_frame_period.value = newsettings.gop;
     v4l2ctl.ApplyControls();
+
+    if (this.config.RTSPServer === 1) {
+      // If it's our RTSPServer, we should have sufficient access to change resolution.
```

Comment:
Actually, I think looking at how it's launched this is probably not true, but I haven't actually tested it.